### PR TITLE
fix: align `_extract_output_tokens` and `AssistantMessageData` validator for non-positive tokens (#837)

### LINKED ISSUE #837: [aw][code health] `_extract_output_tokens` diverges from `AssistantMessageData` validator for negative and zero values despite c
[Content truncated due to length]
## Root Cause

`parser._extract_output_tokens` has a docstring that says it "Replicates the `AssistantMessageData` field-validator behaviour without constructing a full Pydantic model." This claim is incorrect for two cases:

**Case 1 — Zero tokens:**
- `_extract_output_tokens` returns `None` for `outputTokens = 0` (via `tokens if tokens > 0 else None`)
- `AssistantMessageData._sanitize_non_numeric_tokens` passes `0` through unchanged; Pydantic stores `outputTokens = 0`

**Case 2 — Negative tokens:**
- `_extract_output_tokens` returns `None` for `outputTokens = -1` (same guard)
- `AssistantMessageData._sanitize_non_numeric_tokens` only intercepts `bool`/`str` — a negative `int` passes through; Pydantic stores `outputTokens = -1`

These divergences have been present since the fast path was introduced but are currently benign: zero or negative token counts would be unusual in real event files, and `_first_pass` treats `None` and `0` both as "no contribution," which is the correct domain behaviour. However:

1. The docstring creates a false equivalence guarantee, making it easy to update one path without updating the other.
2. Any developer auditing token counts by calling `as_assistant_message()` on events would get different results than `_first_pass` for these edge-case inputs.
3. There is no cross-check test that runs both paths with the same inputs and asserts equivalent behaviour — a future change to either path that introduces additional divergence would not be caught.

## Specific Divergence Table

| Input value | `_extract_output_tokens` | `AssistantMessageData.outputTokens` |
|---|---|---|
| `True` | `None` | `0` (validator maps bool → 0) |
| `False` | `None` | `0` (validator maps bool → 0) |
| `"100"` | `None` | `0` (validator maps str → 0) |
| `0` | `None` | `0` (passes through) |
| `-1` | `None` | `-1` (passes through — no guard) |
| `1234.0` | `1234` | `1234` (Pydantic float→int coercion) |
| `42` | `42` | `42` (passes through) |

## Proposed Resolution

1. **Update `AssistantMessageData._sanitize_non_numeric_tokens`** to also reject non-positive integers (mapping them to `0`). This aligns the model with the fast path's domain intent (negative/zero token counts are meaningless).

2. **Update the `_extract_output_tokens` docstring** to accurately document the intentional differences that remain after the above fix (or confirm equivalence with a comment pointing to the aligned test).

3. **Add a cross-check parameterised test** (`TestExtractOutputTokensEquivalence` or similar) in `tests/copilot_usage/test_parser.py` that runs the same edge-case inputs through both `_extract_output_tokens` and `AssistantMessageData.model_validate({"outputTokens": v}).outputTokens > 0`, asserting that for each input, both paths agree on whether the value contributes a positive token count.

## Testing Requirements

- The existing `TestSanitizeNonNumericTokens` tests in `test_models.py` must be extended to cover `outputTokens = 0` and `outputTokens = -1`, asserting they map to `0`.
- The new cross-check test must cover at minimum: `True`, `False`, `"100"`, `"abc"`, `0`, `-1`, `1`, `1234`, `1234.0`, `3.14`.
- All existing tests must continue to pass without modification.

> Generated by [Code Health Analysis](https://github.com/microsasa/cli-tools/actions/runs/24085974123/agentic_workflow) · ● 4.3M · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsasa%2Fcli-tools%2Fcode-health%22&type=issues)

### PR DESCRIPTION
**Pull request by microsasa**
Closes #837

## Problem

`_extract_output_tokens` (fast path) and `AssistantMessageData._sanitize_non_numeric_tokens` (Pydantic validator) diverged for zero and negative `outputTokens` values:

| Input | `_extract_output_tokens` | `AssistantMessageData.outputTokens` |
|---|---|---|
| `0` | `None` (no contribution) | `0` (passed through) |
| `-1` | `None` (no contribution) | `-1` (passed through) |

While currently benign, the docstring on the fast path claimed equivalence, creating a maintenance hazard.

## Changes

1. **`models.py`** — `_sanitize_non_numeric_tokens` now also rejects non-positive `int`/`float` values (maps to `0`), aligning with the fast path's domain intent that only positive token counts are meaningful.

2. **`parser.py`** — Updated `_extract_output_tokens` docstring to accurately describe the relationship (both paths agree on contribution vs non-contribution; they differ only in representation: `None` vs `0`).

3. **`test_models.py`** — Extended `TestSanitizeNonNumericTokens` with `test_negative_int_maps_to_zero` and `test_large_negative_maps_to_zero`.

4. **`test_parser.py`** — Added `TestExtractOutputTokensEquivalence` cross-check parameterised test covering: `True`, `False`, `"100"`, `"abc"`, `0`, `-1`, `1`, `1234`, `1234.0`, `3.14`. Asserts both paths agree on whether each input contributes positive tokens.

## Verification

- `make check` passes (lint ✅, typecheck ✅, security ✅, 1214 unit tests ✅, 86 e2e tests ✅)
- All existing tests pass without modification (the renamed `test_zero_int_passes_through` → `test_zero_int_maps_to_zero` reflects the new behaviour)

> Generated by [Issue Implementer](https://github.com/microsasa/cli-tools/actions/runs/24091661493/agentic_workflow) · ● 8.2M · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+%22gh-aw-workflow-id%3A+issue-implementer%22&type=pullrequests)

### CODE REVIEW

File: src/copilot_usage/models.py [hunk 1]
```diff
@@ -219,15 +219,20 @@ class AssistantMessageData(BaseModel):
...
+        """Map non-positive and non-numeric token counts to ``0``.
 
-        JSON ``true``/``false`` and numeric strings like ``"100"`` are not
-        valid token counts. Returning ``0`` preserves parsing of the rest of
-        the assistant message payload while preventing these values from
-        being lax-coerced into token counts.
+        JSON ``true``/``false``, numeric strings like ``"100"``, zero, and
+        negative integers are not valid token counts.  Returning ``0``
+        preserves parsing of the rest of the assistant message payload while
+        preventing these values from being lax-coerced into token counts.
```
[the remark this answers is not in the document]
Comment (microsasa) [reply src/copilot_usage/models.py]: Fixed — updated the docstring to say "non-positive numeric values (int/float)" to accurately reflect that the `v <= 0` guard covers both `int` and `float` types, not just integers.

> Generated by [Review Responder](https://github.com/microsasa/cli-tools/actions/runs/24092108953/agentic_workflow) · ● 6M
